respokeSDK/src/main/java/com/digium/respokesdk/RespokeClient.java

```diff
@@ -574,14 +594,20 @@ public void getGroupHistories(final List<String> groupIds, final Integer maxMess
             return;
         }
 
-        Uri.Builder builder = new Uri.Builder();
-        builder.appendQueryParameter("limit", maxMessages.toString());
-        for (String id: groupIds) {
-            builder.appendQueryParameter("groupIds", id);
+        JSONObject body = new JSONObject();
+        try {
+            body.put("limit", maxMessages.toString());
+            JSONArray groupIdParams = new JSONArray(groupIds);
+            body.put("groupIds", groupIdParams);
+        } catch(JSONException e) {
+            getGroupHistoriesError(completionListener, "Error forming JSON body to send.");
+            return;
         }
 
-        String urlEndpoint = "/v1/group-histories" + builder.build().toString();
-        signalingChannel.sendRESTMessage("get", urlEndpoint, null,
+        // This has been modifed to use the newer group-history-search route over the
```

Comment:
May be worth indicating why, as in the kind of errors that can eventually happen.